apps/src/blockly/addons/cdoUtils.js

```diff
@@ -276,11 +263,28 @@ export function getUserTheme(themeOption) {
  * @returns {string} The serialization of the workspace.
  */
 export function getCode(workspace, getSourceAsJson) {
-  if (getSourceAsJson) {
-    return JSON.stringify(Blockly.serialization.workspaces.save(workspace));
-  } else {
+  if (!getSourceAsJson) {
     return Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(workspace));
   }
+
+  const mainWorkspaceSerialization =
+    Blockly.serialization.workspaces.save(workspace);
+
+  const hiddenDefinitionWorkspace = Blockly.getHiddenDefinitionWorkspace();
+  const hiddenWorkspaceSerialization = hiddenDefinitionWorkspace
+    ? Blockly.serialization.workspaces.save(hiddenDefinitionWorkspace)
+    : null;
+
+  // Blocks rendered in the hidden workspace get extra properties that need to be
+  // removed so they don't apply if the block moves to the main workspace on subsequent loads
+  resetEditorWorkspaceBlockConfig(hiddenWorkspaceSerialization.blocks.blocks);
```

Comment:
If shared behaviors are disabled on a level, the serialization may not have any blocks. We should check whether `blocks` exists before trying to perform this operation.